core/src/test/java/org/mule/tck/MuleTestUtils.java

```diff
@@ -596,17 +596,23 @@ public static Flow getTestFlow(String name, MuleContext context) throws Exceptio
     }
 
     @Deprecated
-    public static Service getTestService(String name,
+    public static synchronized Service getTestService(String name,
                                          Class<?> clazz,
                                          Map props,
                                          MuleContext context,
                                          boolean initialize) throws Exception
     {
+        Service service = context.getRegistry().get(name);
```

Comment:
I'm not sure that the intention of this method was to reuse the same service every time